libbeat/outputs/redis/redis_integration_test.go

```diff
@@ -85,6 +85,25 @@ func TestPublishListTLS(t *testing.T) {
 	testPublishList(t, redisConfig)
 }
 
+func TestPublishListMultipleHosts(t *testing.T) {
+	key := "test_publish_tls"
+	db := 0
+	redisConfig := map[string]interface{}{
+		"hosts":    []string{"rediss://" + getSRedisAddr(), "redis://" + getRedisAddr()},
+		"key":      key,
+		"db":       db,
+		"datatype": "list",
+		"timeout":  "5s",
+
+		"ssl.verification_mode": "full",
+		"ssl.certificate_authorities": []string{
+			"../../../testing/environments/docker/sredis/pki/tls/certs/sredis.crt",
+		},
+	}
+
+	testPublishList(t, redisConfig)
+}
```

Comment:
The test via testPublishList will only take the first host configured into account, even if multple are configured. Maybe it's better to introduce some table-driven test like the:

```
func TestWithSchema(t *testing.T) {
  sredisURL := "rediss://" + getSRedisAddr()
  redisURL := "redis://" + getRedisAddr()

  cases := map[string]struct{
    host string
  }{
    "redis ignores ssl settings": {
      host: redisURL,
    },
    "sredis schema sends via tls": {
      host: sredisURL,
    },
  }

  for name, test := range cases {
    t.Run(name, func(t *testing.T) {
      ...
    })
  }
}
```

This adds some more boilerplate, but ensures that we really test with the URLs given, plus gives us a better chance to update/alter the tests in the future.

Reply:
You are right, this is because of https://github.com/elastic/beats/blob/877f1fabeee1da5bc494eab275c16f8c6f64b2e6/libbeat/outputs/redis/redis_integration_test.go#L314

I'll make the changes